src/interfaces/node.cpp

```diff
@@ -83,8 +83,11 @@ class NodeImpl : public Node
     bool shutdownRequested() override { return ShutdownRequested(); }
     void mapPort(bool use_upnp) override
     {
+        // TODO: pass in CConnman and Banman into NodeImpl's constructor
+        if (!g_connman) return;
+
         if (use_upnp) {
-            StartMapPort();
+            StartMapPort(g_connman->GetListenPort());
```

Comment:
In commit 8055e8770a4f43a6381542042d5e54d1a48f54e

This seems fragile. What happens if this interface method is called before `CConnman::Start`?

Reply:
AppInitMain

Reply:
re: https://github.com/bitcoin/bitcoin/pull/14856#discussion_r264335837

I agree with Marco, this is more fragile. It doesn't seem like it should be necessary to rely on connman here just to retrieve a static setting, and there isn't much in the current code organization that would guarantee connman is initialized with the right port number before this call. I think it'd be better to bring back the `GetListenPort()` function and test and call it here and `AppInitMain`. This should also make the diff a little smaller.